Source/Core/Core/PowerPC/PPCTables.cpp

```diff
@@ -55,169 +36,79 @@ const std::array<u64, 16> m_crTable = {{
 
 namespace PPCTables
 {
-GekkoOPInfo* GetOpInfo(UGeckoInstruction inst)
+struct OpDispatch
 {
-  const GekkoOPInfo* info = m_infoTable[inst.OPCD];
-  if (info->type == OpType::Subtable)
-  {
-    switch (inst.OPCD)
-    {
-    case 4:
-      return m_infoTable4[inst.SUBOP10];
-    case 19:
-      return m_infoTable19[inst.SUBOP10];
-    case 31:
-      return m_infoTable31[inst.SUBOP10];
-    case 59:
-      return m_infoTable59[inst.SUBOP5];
-    case 63:
-      return m_infoTable63[inst.SUBOP10];
-    default:
-      ASSERT_MSG(POWERPC, 0, "GetOpInfo - invalid subtable op %08x @ %08x", inst.hex, PC);
-      return nullptr;
-    }
-  }
-  else
-  {
-    if (info->type == OpType::Invalid)
-    {
-      ASSERT_MSG(POWERPC, 0, "GetOpInfo - invalid op %08x @ %08x", inst.hex, PC);
-      return nullptr;
-    }
-    return m_infoTable[inst.OPCD];
-  }
-}
+  std::bitset<64> leaf_flags;
+  std::bitset<64> subtables;
+  u32 op_index_start;
+  u16 dispatch_start;
+  u8 subop_shift;
+  u8 subop_len;
+};
+constexpr std::array<OpDispatch, 30> dispatch_table = {{
+#include "Core/PowerPC/Tables/OpID_DecodingTable.gen.cpp"
+}};
 
-Interpreter::Instruction GetInterpreterOp(UGeckoInstruction inst)
+OpID GetOpID(UGeckoInstruction instruction)
```

Comment:
This feels like a function that could use a few unit tests, but I'd settle for running it vs another disassembler and verifying that they agree on what values are invalid instructions. (2^32 isn't that long to wait.)

Reply:
I don't know how that should be done. I'd be surprised if any disassembler knew what instructions are invalid on the Gekko or whatever the Wii's CPU was called … also, neither this function/table nor the solution that came before it ever checked if the bits that were supposed to be zero (e. g. because they are reserved) actually are zero.